core/trino-main/src/main/java/io/trino/util/StatementUtils.java

```diff
@@ -233,11 +238,60 @@ private StatementUtils() {}
             .build().stream()
             .collect(toImmutableMap(StatementTypeInfo::getStatementType, identity()));
 
-    public static Optional<QueryType> getQueryType(Statement statement)
+    private static final String INFORMATION_SCHEMA = "information_schema";
+    private static final String SYSTEM_JDBC = "system.jdbc";
+    private static final String SYSTEM_METADATA = "system.metadata";
+    private static final String SYSTEM_RUNTIME = "system.runtime";
+
+    private static final ImmutableSet<String> METADATA_CATALOG_SCHEMAS = ImmutableSet.<String>builder()
+            .add(SYSTEM_JDBC)
+            .add(SYSTEM_METADATA)
+            .add(SYSTEM_RUNTIME)
+            .build();
+
+    public static Optional<QueryType> getQueryType(Session session, Statement statement)
     {
         if (statement instanceof ExplainAnalyze) {
-            return getQueryType(((ExplainAnalyze) statement).getStatement());
+            return getQueryType(session, ((ExplainAnalyze) statement).getStatement());
         }
+
+        if (statement instanceof Query) {
+            boolean isMetadataQuery = false;
+
+            HashSet<Table> tables = getTables(statement, new HashSet<String>());
+
+            if (tables.isEmpty()) {
+                isMetadataQuery = true;
+            }
+            else {
+                isMetadataQuery = true;
+                for (Table t : tables) {
+                    String[] tableParts = t.getName().toString().split("\\.");
```

Comment:
You can use `QualifiedName.getParts()` here instead of splitting

Reply:
Made the suggested change.